libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Extensions/RestClient.swift

```diff
@@ -0,0 +1,432 @@
+        let request = self.request(forFileRendition: fileId, version: version, renditionType: type, page: UInt(page), apiVersion: self.apiVersion)
+        self.send(request: request, completionBlock)
+        return request
+    }
+    
+    /// Execute fileRendition request
+    /// - Parameter fileId: An sfdc file identifier.
+    /// - Parameter version: A file version.
+    /// - Parameter type: File type.
+    /// - Parameter page: Page number.
+    /// - Parameter completionBlock: The completion block to invoke.
+    func fileContents(_ fileId: String, version: String, _ completionBlock: @escaping (Result<RestResponse,RestClientError>) -> () ) -> RestRequest {
+        let request = self.request(forFileContents: fileId, version: version, apiVersion: self.apiVersion)
+        self.send(request: request, completionBlock)
+        return request
+    }
+    
+    /// Get File Shares request
+    /// - Parameter sfdcId: Identifier for shares.
+    /// - Parameter page: Page number.
+    /// - Parameter completionBlock: The completion block to invoke.
+    func fileShares(_ sfdcId: String, page: Int, _ completionBlock: @escaping (Result<RestResponse,RestClientError>) -> () ) -> RestRequest {
+       let request = self.request(forFileShares: sfdcId, page: UInt(page), apiVersion: self.apiVersion)
+       self.send(request: request, completionBlock)
+       return request
+    }
+    
+    /// Add a File Share
+    /// - Parameter fileId: File identifier.
+    /// - Parameter entityId: Entity identifier
+    /// - Parameter shareType: The type of Share.
+    /// - Parameter completionBlock: The completion block to invoke.
+    func addFileShare(_ fileId: String, entityId: String, shareType: String, _ completionBlock: @escaping (Result<RestResponse,RestClientError>) -> () ) -> RestRequest {
+       let request = self.request(forAddFileShare: fileId, entityId: entityId, shareType: shareType, apiVersion: self.apiVersion)
+       self.send(request: request, completionBlock)
+       return request
+    }
+    
+    /// Delete a file share.
+    /// - Parameter shareId: Share identtifier.
+    /// - Parameter completionBlock: The completion block to invoke.
+    func deleteFileShare(_ shareId: String, _ completionBlock: @escaping (Result<RestResponse,RestClientError>) -> () ) -> RestRequest {
+       let request = self.request(forDeleteFileShare: shareId, apiVersion: self.apiVersion)
+       self.send(request: request, completionBlock)
+       return request
+    }
+    
+    /// Upload a File.
+    /// - Parameter data: Data contents.
+    /// - Parameter name: Name for the file
+    /// - Parameter description: Description for file.
+    /// - Parameter mimeType: The mime type.
+    /// - Parameter completionBlock: The completion block to invoke.
+    func uploadFileShare(_ data: Data, name: String, description: String, mimeType: String, _ completionBlock: @escaping (Result<RestResponse,RestClientError>) -> () ) -> RestRequest {
```

Comment:
`description` might be optional here, IIRC.

Reply:
In the Obj-c api the description is not optional. Should we change the impl?